packages/react-api/src/api/common.js

```diff
@@ -1,5 +1,13 @@
 import { InvalidColumnError } from '@carto/react-core/';
 
+// Widgets API may return BigInt values, and JSON.parse does not handle them properly, so we need to use a reviver
+const bigIntReviver = (key, value) => {
+  if (typeof value === 'number' && value > Number.MAX_SAFE_INTEGER) {
+    return BigInt(value);
+  }
+  return value;
+};
```

Comment:
See above — unfortunately I think that it's necessary to receive the value as a string and not an integer. By the time the reviver receives a numeric value, I think it will be too late to recover the original integer.

Possible solution in https://stackoverflow.com/a/69644630, though maybe this is complex enough that we want a dependency to help...